experimental/packages/otlp-grpc-exporter-base/src/serializers.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as root from './generated/root';
+import {
+  IExportLogsServiceRequest,
+  IExportLogsServiceResponse,
+  IExportMetricsServiceRequest,
+  IExportMetricsServiceResponse,
+  IExportTraceServiceRequest,
+  IExportTraceServiceResponse,
+} from '@opentelemetry/otlp-transformer';
+import { ExportType } from './internal-types';
+
+const logsResponseType = root.opentelemetry.proto.collector.logs.v1
+  .ExportLogsServiceResponse as ExportType<IExportLogsServiceResponse>;
+
+const logsRequestType = root.opentelemetry.proto.collector.logs.v1
+  .ExportLogsServiceRequest as ExportType<IExportLogsServiceRequest>;
+
+const metricsResponseType = root.opentelemetry.proto.collector.metrics.v1
+  .ExportMetricsServiceResponse as ExportType<IExportMetricsServiceResponse>;
+
+const metricsRequestType = root.opentelemetry.proto.collector.metrics.v1
+  .ExportMetricsServiceRequest as ExportType<IExportMetricsServiceRequest>;
+
+const traceResponseType = root.opentelemetry.proto.collector.trace.v1
+  .ExportTraceServiceResponse as ExportType<IExportTraceServiceResponse>;
+
+const traceRequestType = root.opentelemetry.proto.collector.trace.v1
+  .ExportTraceServiceRequest as ExportType<IExportTraceServiceRequest>;
+
+/**
+ * Serializes and deserializes the OTLP request/response to and from {@link Uint8Array}
+ */
+export interface ISerializer<Request, Response> {
+  serializeRequest(request: Request): Uint8Array | undefined;
+  deserializeResponse(data: Uint8Array): Response;
+}
+
+export const LogsSerializer: ISerializer<
```

Comment:
Yes, I've played with various ways of simplifying this when I added this code but in the end decided on the approach you see here. 

My reasoning is:
- this here is easy to understand
- we don't add signals often, so keeping it like this is likely fine for now

In a follow-up, I'd like to do the following:
- move the proto generation to `@opentelemetry/otlp-transformer`
- have the `@opentelemetry/otlp-transformer` package only export these serializers
- have the serializers take the internal representation from the SDK as an input, instead of an intermediate representation, which will allow us more flexibility in getting rid of that intermediate representation without it being a breaking change after we release the exporters as GA

Reply:
> we don't add signals often, so keeping it like this is likely fine for now

makes sense